enos/enos-scenario-replication.hcl

```diff
@@ -0,0 +1,682 @@
+      arch                 = matrix.artifact_source == "artifactory" ? matrix.arch : null
+      product_version      = var.vault_product_version
+      artifact_type        = matrix.artifact_type
+      distro               = matrix.artifact_source == "artifactory" ? matrix.distro : null
+      edition              = matrix.artifact_source == "artifactory" ? matrix.edition : null
+      instance_type        = matrix.artifact_source == "artifactory" ? local.vault_instance_type : null
+      revision             = var.vault_revision
+    }
+  }
+
+  step "find_azs" {
+    module = module.az_finder
+    variables {
+      instance_type = [
+        local.vault_instance_type
+      ]
+    }
+  }
+
+  step "create_vpc" {
+    module     = module.create_vpc
+    depends_on = [step.find_azs]
+
+    variables {
+      ami_architectures  = [matrix.arch]
+      availability_zones = step.find_azs.availability_zones
+      common_tags        = local.tags
+    }
+  }
+
+  step "read_license" {
+    module = module.read_license
+
+    variables {
+      file_name = abspath(joinpath(path.root, "./support/vault.hclic"))
+    }
+  }
+
+  step "create_primary_backend_cluster" {
+    module     = "backend_${matrix.primary_backend}"
+    depends_on = [step.create_vpc]
+
+    providers = {
+      enos = provider.enos.ubuntu
+    }
+
+    variables {
+      ami_id      = step.create_vpc.ami_ids["ubuntu"][matrix.arch]
+      common_tags = local.tags
+      consul_release = {
+        edition = var.backend_edition
+        version = matrix.consul_version
+      }
+      instance_type = var.backend_instance_type
+      kms_key_arn   = step.create_vpc.kms_key_arn
+      vpc_id        = step.create_vpc.vpc_id
+    }
+  }
+
+  step "create_vault_primary_cluster" {
+    module = module.vault_cluster
+    depends_on = [
+      step.create_primary_backend_cluster,
+      step.build_vault,
+    ]
+    providers = {
+      enos = local.enos_provider[matrix.distro]
+    }
+
+    variables {
+      ami_id             = step.create_vpc.ami_ids[matrix.distro][matrix.arch]
+      common_tags        = local.tags
+      consul_cluster_tag = step.create_primary_backend_cluster.consul_cluster_tag
+      consul_release = matrix.primary_backend == "consul" ? {
+        edition = var.backend_edition
+        version = matrix.consul_version
+      } : null
+      dependencies_to_install   = local.dependencies_to_install
+      instance_type             = local.vault_instance_type
+      kms_key_arn               = step.create_vpc.kms_key_arn
+      storage_backend           = matrix.primary_backend
+      unseal_method             = matrix.primary_seal
+      vault_local_artifact_path = local.bundle_path
+      vault_install_dir         = local.vault_install_dir
+      vault_artifactory_release = matrix.artifact_source == "artifactory" ? step.build_vault.vault_artifactory_release : null
+      vault_environment = {
+        VAULT_LOG_LEVEL = "debug"
```

Comment:
🤔 we might want to make this exposed via a variable (thinking ahead for longer-running scenarios, this could fill up our tiny default sized disks)